cloud/aws/actuators/machine/actuator.go

```diff
@@ -92,6 +102,37 @@ func (a *Actuator) Create(cluster *clusterv1.Cluster, machine *clusterv1.Machine
 	return a.updateStatus(machine, instance, mLog)
 }
 
+// updateMachineProviderConditions updates conditions set within machine provider status.
+func (a *Actuator) updateMachineProviderConditions(machine *clusterv1.Machine, mLog log.FieldLogger, conditionType providerconfigv1.AWSMachineProviderConditionType, reason string, msg string) error {
+
+	mLog.Debug("updating machine conditions")
+
+	awsStatus, err := AWSMachineProviderStatusFromClusterAPIMachine(machine)
+	if err != nil {
+		mLog.Errorf("error decoding machine provider status: %v", err)
+		return err
+	}
+
+	awsStatus.Conditions = SetAWSMachineProviderCondition(awsStatus.Conditions, conditionType, corev1.ConditionTrue, reason, msg, UpdateConditionIfReasonOrMessageChange)
+
+	awsStatusRaw, err := EncodeAWSMachineProviderStatus(awsStatus)
+	if err != nil {
+		mLog.Errorf("error encoding AWS provider status: %v", err)
+		return err
+	}
+
+	machineCopy := machine.DeepCopy()
```

Comment:
Let's:

```go
func (a *Actuator) updateMachineStatus(machine *clusterv1.Machine, awsStatus *providerconfigv1.AWSMachineProviderStatus, mLog log.FieldLogger, networkAddresses []corev1.NodeAddress{}) error {
 	awsStatusRaw, err := EncodeAWSMachineProviderStatus(awsStatus)
 	if err != nil {
 		mLog.Errorf("error encoding AWS provider status: %v", err)
 		return err
 	}

 	machineCopy := machine.DeepCopy()
 	machineCopy.Status.ProviderStatus = awsStatusRaw
        if networkAddresses != nil {
 		machineCopy.Status.Addresses = networkAddresses
	}
 
 	if !equality.Semantic.DeepEqual(machine.Status, machineCopy.Status) {
 		mLog.Info("machine status has changed, updating")
 		machineCopy.Status.LastUpdated = metav1.Now()
 
 		_, err := a.clusterClient.ClusterV1alpha1().Machines(machineCopy.Namespace).UpdateStatus(machineCopy)
 		if err != nil {
 			mLog.Errorf("error updating machine status: %v", err)
 			return err
 		}
 	} else {
 		mLog.Debug("status unchanged")
 	}
	return nil
}
```

then:

```go
	a.updateMachineStatus(machine, awsStatus, nil, mLog)
```

and on the lines `573:594` then replace the code with:

```go
	a.updateMachineStatus(machine, awsStatus, networkAddresses, mLog)
```

so only one method is responsible for updated machine's status.